library/clinical_rules.php

```diff
@@ -220,6 +216,8 @@ function active_alert_summary($patient_id, $mode, $dateTarget = '', $organize_mo
     $actions = test_rules_clinic('', 'active_alert', $dateTarget, $mode, $patient_id, '', $organize_mode, array(), 'primary', null, null, $user);
 
     if (empty($actions)) {
+        // when there are no actions we have to update alert_notify_pid session variable
+        OpenEMR\Common\Session\SessionUtil::setSession('alert_notify_pid', $patient_id);
```

Comment:
Just make sure this't doesn't create a 'PHP Warning: session_start(): Cannot start session when headers already sent.' message, which happens if try to open a session after any html has already been sent.

Reply:
The logic on this is also odd. I think the point of the alert_notify_pid session variable is to just ensure that the active alert does not keep on popping up every time the patient summary screen is refreshed. To do this, it only shows when the pid changes. So not sure why it needs to be updated when no actions. This is why in 5.0.3, removed this and just simply updated it in demographics.php after the popup showed up.

Reply:
oh wait, I meant 6.0.0 :)

Reply:
I had User Debug turn on so warning were suppressed. It works okay in 6.0.0 though not 502 demo fix.
I'll remove in 6 and 502

Reply:
think i discovered that when you click on a new patient and there are no actions it doesn't update the current patient correctly

Reply:
are you using the last commit I made?

Reply:
no, sorry, was referring to 
>The logic on this is also odd. I think the point of the alert_notify_pid session variable is to just ensure that the active alert does not keep on popping up every time the patient summary screen is refreshed. To do this, it only shows when the pid changes. So not sure why it needs to be updated when no actions.

Reply:
oh, I see. That makes sense since this gets called before the reset in demographics.
I can directly set the session local and it will be in scope when it hits the demographics popup so maybe we'll put this back in....

Reply:
Testing well on PHP 72 and 7.4. Bringing in so we can test on demo servers.
Leaving alert reset alone as I can't find a reason for it to be where it is at in clinical reminders.